rfc/rfc-2/2 - Deletion Info Conversion.md

```diff
@@ -0,0 +1,221 @@
+not only contain data files but also deletion vectors. The consumers which cannot handle instance of 
+`InternalDeletionVector` in `filesAdded` can either ignore them or raise an exception.
+
+**Updates in `DeltaActionsConverter`**
+The converter class is responsible for converting Delta Lake `actions` into internal representations, resulting in
+`TableChange` objects. The converter is updated to extract deletion vectors from the `AddFile` actions. Currently,
+the converters read content from source metadata files and then close the file handles. However, this will be different
+for deletion vectors. To avoid the potential memory explosion, the converters will create a provider of deletion vectors
+ordinals, which will be used to stream the ordinals of the records deleted to the targets.
+
+**Changes in `DeltaConversionSource`**
+As described above, the delta commit logs adds data file entries in commit log each time delete vectors are updated. 
+This needs special handling in the delta source, to avoid duplication of data file actions.
+
+**New class `IcebergDeleteVectorConverter`**
+
+The purpose of this class is to convert internal deletion vectors to Iceberg positional delete files. The class will
+contain methods matching other Iceberg converters, for instance `toIceberg`. However, similar to unique handling of
+deletion vectors in `DeltaActionsConverter`, this class will have a unique implementation. This class will generate 
+data files for deletion vectors, which will be added to the target table. The cost of conversion of deletion vectors
+is proportional to the number of records deleted, and not the size of metadata or number of files. 
+
+Iceberg is quite
+flexible when it comes to location of data and metadata files. Since deletion vectors are data files, colocating them
+with metadata files could be confusing. Typically, deletion files are stored in partition specific directories, in
+the data directory of a table. However, it seems cleaner to keep all XTable generated data files in a separate
+directory, which will be configurable. The default location for deletion vectors will be `deletion-vectors` directory
+in the base directory of the table.
+
+**Modifications in `IcebergDataFileUpdatesSync` and `IcebergConversionTarget`**
+Finally, existing classes that handle Iceberg data files and sync operations will be updated to handle deletion vectors.
+Most changes are straightforward and inline with existing logic. However, as described earlier, new deletion vectors
+are added to a separate manifest file in Iceberg. This requires invoking a separate transaction operation specific 
+to row level operations. The row level operation is initiated only if there are new deletion vectors in the 
+`FilesAdded`.
+
+## Rollout/Adoption Plan
+
+This change does not cause any breaking changes. Before this change, delete vectors were ignored and would cause
```

Comment:
Looks like the proposal is to only implement V2 positional deletes. Is that correct? Should that be added here?

Reply:
Hi @vigneshc 
Thanks for reviewing the proposal! 

Iceberg V3 hasn't been released yet. The good news is that its implementation aligns with Delta Lake, but it will be a bit early to dive into the details and implementation for V3 conversions in this RFC. There's also another active thread on modularizing XTable code, with plans to have separate modules for V2 and V3. 

I hope I got your question right?

Reply:
Thanks Ashvin. I was asking to explicitly add that the rfc currently only covers V2 spec.

Reply:
The RFC is recommending a generic model that allows us to evolve alongside the different table formats. The V2 spec is mainly in the RFC to illustrate why this is the recommended approach, so I don't know if we need to explicitly call this out in the RFC. We can add the versions support in our docs after we get a first iteration completed.